### PR TITLE
feat(ihev2): fix for allscripts auth and athena sha1

### PR DESCRIPTION
**Pull request by jonahkaye**
Refs: #[1667](https://github.com/metriport/metriport-internal/issues/1667)

### Description

- fixes for allscripts and athena

### Testing

- Local
  - [x] tested with prod endpoints

### Release Plan

- :warning: Points to `master`
- turn on ihev2 fully for (all?) customers 
- monitor analytics: we shouldnt have an event lasting greater than 300000 ms again  
- [ ] Merge this

### CODE REVIEW

File: packages/core/src/external/carequality/ihe-gateway-v2/saml/security/sign.ts [hunk 1]
```diff
@@ -25,11 +25,11 @@ function createSignature({
   const sig = new SignedXml({ privateKey });
   sig.addReference({
     xpath: xpath,
-    digestAlgorithm: "http://www.w3.org/2000/09/xmldsig#sha1",
+    digestAlgorithm: "http://www.w3.org/2001/04/xmlenc#sha256",
     transforms: transforms,
   });
   sig.canonicalizationAlgorithm = "http://www.w3.org/2001/10/xml-exc-c14n#";
-  sig.signatureAlgorithm = "http://www.w3.org/2000/09/xmldsig#rsa-sha1";
+  sig.signatureAlgorithm = "http://www.w3.org/2001/04/xmldsig-more#rsa-sha256";
```
Comment (jonahkaye) [inline packages/core/src/external/carequality/ihe-gateway-v2/saml/security/sign.ts:32]: athena doesnt like sha1. Tested sha256 against all endpoints and compatible

File: packages/core/src/external/carequality/ihe-gateway-v2/saml/saml-client.ts [hunk 2]
```diff
@@ -51,8 +51,9 @@ async function getTrustedKeyStore(): Promise<string> {
 export function setRejectUnauthorized(value: boolean): void {
   rejectUnauthorized = value;
 }
+// TODO: remove false here once allscripts patches their endpoints
 export function getRejectUnauthorized(): boolean {
-  return rejectUnauthorized;
+  return false && rejectUnauthorized;
```
Comment (jonahkaye) [inline packages/core/src/external/carequality/ihe-gateway-v2/saml/saml-client.ts:56]: always false for now. Will change this back to true once allscripts fixes this

**Review by Goncharo — APPROVED** [review]
:shipit:

File: packages/core/src/external/carequality/ihe-gateway-v2/saml/saml-client.ts [hunk 3]
```diff
@@ -51,8 +51,9 @@ async function getTrustedKeyStore(): Promise<string> {
 export function setRejectUnauthorized(value: boolean): void {
   rejectUnauthorized = value;
 }
+// TODO: remove false here once allscripts patches their endpoints
```
Comment (Goncharo) [inline packages/core/src/external/carequality/ihe-gateway-v2/saml/saml-client.ts:54]: @jonahkaye pls tie this TODO to an issue # so we don't forget

File: packages/core/src/external/carequality/ihe-gateway-v2/saml/security/sign.ts (hunk 1 above)
Comment (jonahkaye) [reply packages/core/src/external/carequality/ihe-gateway-v2/saml/security/sign.ts:32]: Seems not compatible with stargate. Investigating

**Pull request merged**